APHKLogicExtractor/Loader/MaybeFileConverter.cs

```diff
@@ -0,0 +1,103 @@
+using System.Reflection;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using RandomizerCore.Json;
+
+namespace APHKLogicExtractor.Loader;
+
+public class MaybeFileConverter(ResourceLoader resourceLoader) : JsonConverter
+{
+    static readonly Type target = typeof(MaybeFile<>).GetGenericTypeDefinition();
+
+    public override bool CanConvert(Type objectType)
+    {
+        return objectType.IsGenericType && objectType.GetGenericTypeDefinition() == MaybeFileConverter.target;
+    }
+
+    public override object? ReadJson(
+        JsonReader reader,
+        Type objectType,
+        object? existingValue,
+        JsonSerializer serializer)
+    {
+        // The token is a string, therefore we'll try to load using the resourceLoader.
+        if (reader.TokenType == JsonToken.String)
+        {
+            var instance = Activator.CreateInstance(objectType);
+            objectType.GetProperty("Lazy")?.SetValue(instance, (resourceLoader, reader.Value as string));
+            return instance;
+        }
+
+        // We detected an object or array in the JSON, therefore, we'll try to parse as usual.
+        if (reader.TokenType == JsonToken.StartObject || reader.TokenType == JsonToken.StartArray)
+        {
+            // Grabs `MaybeFile` generic argument `T`
+            var innerType = objectType.GetGenericArguments()[0];
+
+            var deserialized = serializer.Deserialize(reader, innerType);
+            var instance = Activator.CreateInstance(objectType);
+            objectType.GetProperty(nameof(MaybeFile<object>.Content))?.SetValue(instance, deserialized);
+            if (innerType == typeof(JToken))
+                objectType.GetProperty(nameof(MaybeFile<object>.Serializer))?.SetValue(instance, serializer);
+            return instance;
+        }
+
+        throw new Exception($"Unable to deserialize token {reader.TokenType}.");
+    }
+
+    public override void WriteJson(
+        JsonWriter writer,
+        object? value,
+        JsonSerializer serializer)
+    {
+        var objType = value?.GetType();
```

Comment:
We shouldn't need this check since newtonsoft will only call our serializer when CanConvert has returned true

Reply:
Fixed